### PR TITLE
Configured the DD units to use the consolidated image

### PR DESCRIPTION
**Pull request by alexcristi**
## Changelog
Unified all DD agents into a single one: https://github.com/adobe-platform/ethos-dd-agent
The new image is: __index.docker.io/adobeplatform/ethos-dd-agent:latest__

The checks are configured based on a new env variable:
- __DD_TIER__ (control | worker | proxy)

## Notes

Did you introduce any command that executes `docker run`?
R: No

If you did, have you ensured that all your `docker run` using the default logger have either `max-size` and/or `max-file` set?
R: N/A

Are there any dependencies on github.com/adobe-platform/infrastructure?
R: No

If so:
 
Is an update to the base stack required(rare)?
R: N/A

Is an A/B required?
R: N/A

Are you dependent on new secrets, or infrastructure in any way?
R: N/A

### CODE REVIEW

**Comment by bossjones** [comment]
Looks great, yay for unified solutions!

**Comment by bossjones** [comment]
👍

**Comment by digitalyuki** [comment]
Perhaps it’s time to rename the datadog container names…
this is our current state:

`datadog-control.service` (runs container named `dd-agent-mesos` on a single control node)
`datadog-mesos-master.service` (runs container named `dd-agent-mesos-master` on all control nodes)
`datadog-mesos-slave.service` (runs container named `dd-agent-mesos-slave` on all worker nodes)
`datadog-proxy.service` (runs container named `dd-agent-proxy` on all proxy nodes)

it’s a bit confusing. 
@doalexan: I'm wondering if the remaining checks in the `dd-agent-mesos` container can be moved to run on each control node, if that’s the case then we can just consolidate those two control-tier-based agents, then we could have:

`datadog-control.service` (runs container named `dd-agent-control` on all control nodes)
`datadog-worker.service` (runs container named `dd-agent-worker` on all worker nodes)
`datadog-proxy.service` (runs container named `dd-agent-proxy` on all proxy nodes)

**Comment by alexcristi** [comment]
@digitalyuki @matthewdfuller: Integrated the feedback.

File: v1/opt/datadog/setup/leader/001-images.sh [hunk 1]
```diff
@@ -8,3 +8,6 @@ etcd-set /images/dd-agent-mesos         "index.docker.io/behance/docker-dd-agent
 etcd-set /images/dd-agent-mesos-master  "index.docker.io/adobeplatform/docker-dd-agent-mesos-master:latest"
 etcd-set /images/dd-agent-mesos-slave   "index.docker.io/adobeplatform/docker-dd-agent-mesos-slave:latest"
 etcd-set /images/dd-agent-proxy         "index.docker.io/behance/docker-dd-agent-proxy:latest"
+
+# TODO: remove the keys above as they are not needed anymore
+etcd-set /images/ethos-dd-agent         "index.docker.io/adobeplatform/ethos-dd-agent:latest"
```
Comment (matthewdfuller) [inline v1/opt/datadog/setup/leader/001-images.sh]: This is the spot - can you change `latest` to a specific tag?

Comment (alexcristi) [inline v1/opt/datadog/setup/leader/001-images.sh]: Absolutely, will do this asap.

**Comment by matthewdfuller** [comment]
@alexcristi one last thing - can you please update your `docker run` lines to match those that @ts3ng has in his PR that I just merged? (https://github.com/adobe-platform/ethos-systemd/pull/21). This will fix the merge conflict.

**Comment by matthewdfuller** [comment]
Thanks for adding those - it looks like there is still a merge conflict though.

**Comment by alexcristi** [comment]
Solved.

**Pull request merged**